### ISSUE TITLE
tsql LT01 bug. `GO` statement must stay on its own line

### ISSUE THREAD

**Issue by fmms** [sqlfluff/sqlfluff#5978]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I tried to format a tsql file and `sqlfluff` broke the codebase

### Expected Behaviour

Code is formatted and can still be executed.

### Observed Behaviour

Cannot execute the code anymore.

### How to reproduce

create a file query:

```sql
DROP FUNCTION IF EXISTS INTERNAL_ETL.DIL_md_ScdTest
GO
DROP FUNCTION IF EXISTS INTERNAL_ETL.DIL_md_ScdTest
GO
```

Then run the following command:
```ps1
PS C:\Users\fmoeller\Desktop\RepositoryVorlage>  sqlfluff fix --dialect tsql  .\query.sql
==== finding fixable violations ====
== [query.sql] FAIL
L:   1 | P:  52 | LT01 | Unexpected line break. [layout.spacing]
L:   3 | P:  52 | LT01 | Unexpected line break. [layout.spacing]
== [query.sql] FIXED
2 fixable linting violations found
```

The output is 

```sql
DROP FUNCTION IF EXISTS INTERNAL_ETL.DIL_md_ScdTestGO
DROP FUNCTION IF EXISTS INTERNAL_ETL.DIL_md_ScdTestGO
```

`GO` or `GO <n>` must always be on its own line.

### Dialect

tsql

### Version

```ps1
PS C:\Users\fmoeller\Desktop\RepositoryVorlage> sqlfluff version -v
==== sqlfluff ====
sqlfluff:                3.0.7 python:                 3.11.9
implementation:        cpython verbosity:                   1
dialect:                  tsql templater:                 raw
rules:                                all
```

### Configuration

none

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by keraion** [comment]
Related #5808
